### DISCUSSION TITLE
native controller support

### DISCUSSION THREAD

**Discussion by JunkyardCat** [LizardByte/community#506] [Sunshine]
is it possible to integrate controller support without having to install vigem driver (whatever its named) to make it work

**Comment by ReenigneArcher** [comment]
Unfortunately we need a way to create virtual hardware drivers. This requires a special type of signing certificate which is only available to legal businesses (for which we are not one right now). So we rely on vigembus to do this for us.

**Comment by JunkyardCat** [comment]
@ReenigneArcher 
is there a way to do it like how steam link/steam does it? so when i use steam link i can use a controller also it can accept the touch screen controls too

**Comment by f8ith** [comment]
[EmuController](https://github.com/FirstPlatoLV/EmuController)

This project is a POC of creating Virtual HID devices in user-mode. User-mode drivers currently do not require signing, so we could use user-mode frameworks like [VirtualHidMini](https://github.com/microsoft/DMF/blob/master/Dmf/Modules.Library/Dmf_VirtualHidMini.md) to provide controller support without ViGEMBus.

**Reply by Billybishop** [reply]
#652 @ReenigneArcher
